### PR TITLE
Feature/protein sequences

### PR DESCRIPTION
**Pull request by azurebrd**
Hi @sibyl229 ,

This is returning what looks to me like the correct JSON.  Yesterday I could test it on 
https://juancarlos.wormbase.org/species/c_elegans/protein/WP:CE48339#067--10
but today, it redirects to 
https://juancarlos.wormbase.org/search/protein/WP:CE48339?redirect=1#067--10
which says "Sorry, 0 results found for the search WP:CE48339"  

Do you know if something's wrong with the search server ?  I haven't changed anything since yesterday, nor since you fixed the wormbase_local.conf for me.

This probably shouldn't get merged until we can see that it looks correct on a webpage.

Thanks !

### CODE REVIEW

**Comment by sibyl229** [comment]
HI @azurebrd I think the URL might be: https://juancarlos.wormbase.org/species/c_elegans/protein/CE48339#067--10 due to identifier change. Does that work?

**Comment by sibyl229** [comment]
And be sure to use the WS266 database.

File: src/rest_api/classes/protein/widgets/sequences.clj [hunk 1]
```diff
@@ -1,21 +1,61 @@
...
+   "F" "Phe"
+   "O" "Pyl*"
+   "P" "Pro"
+   "S" "Ser"
+   "T" "Thr"
+   "W" "Trp"
+   "Y" "Tyr"
+   "V" "Val"
+   "U" "Sec*"
+   "X" "Xaa**"})
```
Comment (sibyl229) [inline src/rest_api/classes/protein/widgets/sequences.clj:33]: Out of curiosity, why do some amino acids have * in their names?

File: src/rest_api/classes/protein/widgets/sequences.clj [hunk 2]
```diff
@@ -1,21 +1,61 @@
...
            (format "%.1f" mw))
    :description "the estimated molecular weight of the protein"})
 
 (defn amino-acid-composition [p]
-  {:data (keys p)
-   :d (:db/id p) ; bioperl uses seqStats
+  {:data 
+   (let [source (when-let [h (:protein/peptide p)]
+                  (->(:peptide/sequence (:protein.peptide/peptide h))
+                                        (str/split  #"")
```
Comment (sibyl229) [inline src/rest_api/classes/protein/widgets/sequences.clj]: Some indentation is a bit off here and in a few other place. Can you configure your text editor to reformat the code using indentation rules for Clojure?

Here is the style guide: https://github.com/bbatsov/clojure-style-guide

Comment (sibyl229) [reply src/rest_api/classes/protein/widgets/sequences.clj]: If you have problem setting it up, we can talk about it and figure something out.

File: src/rest_api/classes/protein/widgets/sequences.clj (hunk 1 above)
Comment (azurebrd) [reply src/rest_api/classes/protein/widgets/sequences.clj:33]: I don't know :)  I got that from the acedb code on the website repo website/lib/WormBase/API/Object/Protein.pm

File: src/rest_api/classes/protein/widgets/sequences.clj (hunk 2 above)
Comment (azurebrd) [reply src/rest_api/classes/protein/widgets/sequences.clj]: Ah, that's weird.  Adam helped me set up vim to auto-indent by highlighting everything and pressing = .  I think the lack of a space between (-> and (:peptide/sequence was messing that up.

I've committed another change for "Indentation changes" to the repo

**Comment by azurebrd** [comment]
Hi @sibyl229 

> I think the URL might be: https://juancarlos.wormbase.org/species/c_elegans/protein/CE48339#067--10 due to identifier change. Does that work?

Oh, I see, the WP: is gone.  Cool, that loads a page.  The widget doesn't work though.

> And be sure to use the WS266 database.

Where can I check that I'm doing that ?

Yesterday the widget was loading all the other data.  Today, even if I remove the code I added, I'm getting a : 

Server Error: 500 at rest/widget/protein/CE48339/sequences

Error Message:
Caught exception in WormBase::Web::Controller::REST->widget "http://ec2-18-206-182-237.compute-1.amazonaws.com:4001/rest/widget/protein/CE48339/sequences failed with 404 at /mnt/data1/user-data/jchan/website/script/../lib/WormBase/Web/Controller/REST.pm line 923." 

This seems like a different issue to me, but I don't know.  Do you know what could be causing this ?

Thanks !

**Comment by sibyl229** [comment]
Hi @azurebrd the REST API error is due to wrong version of the database being used. Please ensure that the environment variable`WB_DB_URI=datomic:ddb://us-east-1/WS266/wormbase` is set. It ensures that the right database is used.

**Comment by azurebrd** [comment]
Got it, thanks you @sibyl229 !

The sequence field from curling the JSON looks the same to me as in wormbase, but comparing the webpages : 
https://juancarlos.wormbase.org/species/c_elegans/protein/CE48339#6--10
https://wormbase.org/species/c_elegans/protein/WP:CE48339#6--10

The one I'm working on says "No sequence data available for download".  Do you know how that download button works, and whether something might have changed between releases that needs updating ?  From what Adam showed me, I just try to make the JSON match the perl version of the JSON, and these match (other than the actual value being slightly different, which I'm assuming is from it being a different release).

Thanks again !

**Comment by azurebrd** [comment]
Wow, thanks @sibyl229 sorry about that typo.

I've tested it, and it looks good to me.  If you like we can pull it, or ask Adam + Matt for review.

**Comment by sibyl229** [comment]
Hi @khowe do you know if we can _always_ get the protein sequence from Datomic (`(:peptide/sequence (:protein.peptide/peptide x)`) or do we ever have to translate from sequences in the sequence database? Thanks!

**Comment by khowe** [comment]
I believe that all Protein objects that we need peptide sequence on the site should have the peptide sequence directly attached. @epaule ?

**Comment by epaule** [comment]
they should. If they are not, we need to have a look into it.

**Comment by sibyl229** [comment]
Great! Thank you @khowe @epaule !

**Comment by epaule** [comment]
just looked and some of the history CDSes have proteins without peptides

**Comment by sibyl229** [comment]
Hi @azurebrd I merged your PR for protein sequences. And it's deployed on staging for testing.

**Comment by sibyl229** [comment]
@epaule do we need to worry about those proteins of history CDSes?

**Comment by epaule** [comment]
also any non-elegans proteins that have been withdrawn/changed by the respective source but exist in historical homology groups don't have peptides

**Comment by sibyl229** [comment]
@epaule should those protein sequences be displayed? Does the live site show those sequences? @azurebrd 's new datomic-based implementation is currently deployed to staging, so it's probably missing those sequences.

**Comment by epaule** [comment]
I think it is fine to not display the protein sequence of historic proteins
that don't exist anymore.

**Comment by sibyl229** [comment]
@epaule great! Thanks for clarifying

**Comment by azurebrd** [comment]
@sibyl229 awesome, thanks !

**Pull request merged**